libimage/manifest_list.go

```diff
@@ -325,8 +325,19 @@ func (m *ManifestList) Add(ctx context.Context, name string, options *ManifestLi
 	if err != nil {
 		withDocker := fmt.Sprintf("%s://%s", docker.Transport.Name(), name)
 		ref, err = alltransports.ParseImageName(withDocker)
+		if err == nil {
```

Comment:
looks like this whole block up until the LookupImage part is the same in all the places where you touched this so I think it would make sense to move this into a helper function to avoid the duplication

Reply:
Factored out.